File: 0387-first-unique-character-in-a-string/0387-first-unique-character-in-a-string.js
/**
 * @param {string} s
 * @return {number}
 */
var firstUniqChar = function(s) {
    const freq = new Array(26).fill(0);
    const aCode = 'a'.charCodeAt(0);
    
    for (let i = 0; i < s.length; i++) {
        freq[s.charCodeAt(i) - aCode]++;
    }
    
    for (let i = 0; i < s.length; i++) {
        if (freq[s.charCodeAt(i) - aCode] === 1) {
            return i;
        }
    }
    
    return -1;
};

File: 0387-first-unique-character-in-a-string/0387-first-unique-character-in-a-string.cpp
class Solution {
public:
    int firstUniqChar(string s) {
        vector<int> freq(26, 0);
        int aCode = 'a';

        for (int i = 0; i < s.length(); i++) {
            freq[s[i] - aCode]++;
        }

        for (int i = 0; i < s.length(); i++) {
            if (freq[s[i] - aCode] == 1) return i;
        }
        
        return -1;
    }
};